### PR TITLE
[dunfell] Regenerate all recipes with modified superflore respecting conditionals in <export><build_type>

### PR DESCRIPTION
**Pull request by shr-project**
Superflore from this PR:
https://github.com/ros-infrastructure/superflore/pull/280

### CODE REVIEW

File: scripts/ros-generate-cache.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@
 # Copyright (c) 2019-2020 LG Electronics, Inc.
 
 readonly SCRIPT_NAME="ros-generate-cache"
-readonly SCRIPT_VERSION="1.5.3"
+readonly SCRIPT_VERSION="1.6.0"
```
Comment (herb-kuta-lge) [inline scripts/ros-generate-cache.sh:21]: Commit body:

> leave ROS_VERSION and ROS_PYTHON_VERSION setting for
>  "conditional" attributes on superflore itself

I don't understand what this means.

File: scripts/ros-generate-recipes.sh [hunk 2]
```diff
@@ -47,17 +47,14 @@ if [ -z "$SUPERFLORE_GEN_OE_RECIPES" ]; then
 fi
 
 ROS_DISTRO=$1
-# ROS_VERSION and ROS_PYTHON_VERSION must be in the environment as they appear in "conditional" attributes. Keep this block in
-# sync with the one in ros-generate-cache.sh .
+# Keep this block in sync with the one in ros-generate-cache.sh .
 case $ROS_DISTRO in
     "melodic"|"noetic")
         export ROS_VERSION="1"
```
Comment (herb-kuta-lge) [inline scripts/ros-generate-recipes.sh:53]: > ROS_VERSION (not exported)

It's still being exported in this script.

**Review by herb-kuta-lge — CHANGES_REQUESTED** [review]

File: scripts/ros-generate-cache.sh (hunk 1 above)
Comment (herb-kuta-lge) [inline scripts/ros-generate-cache.sh:21]: Commit body:

> leave ROS_VERSION and ROS_PYTHON_VERSION setting for
>  "conditional" attributes on superflore itself

I don't understand what this means.

File: scripts/ros-generate-recipes.sh (hunk 2 above)
Comment (herb-kuta-lge) [inline scripts/ros-generate-recipes.sh]: > ROS_VERSION (not exported)

It's still being exported in this script.

**Review by herb-kuta-lge — CHANGES_REQUESTED** [review]

File: scripts/ros-generate-cache.sh (hunk 1 above)
Comment (shr-project) [reply scripts/ros-generate-cache.sh:21]: I meant that we'll leave it on this superflore implementation:
https://github.com/ros-infrastructure/superflore/pull/280/commits/5b9014b14c50e6a79c6bce4e5c8de28b4bb270b4
to set ROS_VERSION and ROS_PYTHON_VERSION based on the distribution name

File: scripts/ros-generate-recipes.sh (hunk 2 above)
Comment (shr-project) [reply scripts/ros-generate-recipes.sh]: Yes, but it doesn't need to be exported I believe.

Comment (herb-kuta-lge) [reply scripts/ros-generate-recipes.sh]: So shouldn't `export` be dropped?

File: scripts/ros-generate-cache.sh (hunk 1 above)
Comment (herb-kuta-lge) [reply scripts/ros-generate-cache.sh:21]: In that case:

`don't set ROS_PYTHON_VERSION (v1.6.0)` -> `don't export ROS_VERSION &  ROS_PYTHON_VERSION (v1.6.0)`

`leave ROS_VERSION and ROS_PYTHON_VERSION setting for "conditional" attributes on superflore itself` -> `superflore no longer reads ROS_VERSION and ROS_PYTHON_VERSION from its environment but derives them from the specified ROS_DISTRO, so don't export them.`

**Review by herb-kuta-lge — CHANGES_REQUESTED** [review]

**Review by herb-kuta-lge — APPROVED** [review]

**Pull request merged**